### ISSUE TITLE
Mismatch in schema name validation

### ISSUE THREAD

**Issue by antongakh** [angular/angular-cli#25704]
### Command

new

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

There is a possibility to create an application with the `ng new` command with the name which makes it impossible to configure it through the `ng config` command.

### Minimal Reproduction

1. generate new application with `ng new my.application`
2. run any `ng config` command in generated application (e.g. `ng config schematics.@schematics/angular:component.style scss`)
3. Observe command output

ER: Command executed correctly
AR: Error:
`Error: Schema validation failed with the following errors:
  Data path "/projects" must NOT have additional properties(my.application).`

### Exception or Error

```text
Error: Schema validation failed with the following errors:
  Data path "/projects" must NOT have additional properties(my.application)
```

### Your Environment

```text
Angular CLI: 16.2.0
Node: 18.16.0
Package Manager: npm 9.5.1
OS: win32 x64

Angular: undefined
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1602.0
@angular-devkit/core         16.2.0
@angular-devkit/schematics   16.2.0
@angular/cli                 16.2.0
@schematics/angular          16.2.0
```

### Anything else relevant?

Application name is validated by: https://github.com/angular/angular-cli/blob/main/packages/schematics/angular/application/schema.json#L16
Project name is validated by: https://github.com/angular/angular-cli/blob/main/packages/angular/cli/lib/config/workspace-schema.json#L26C9-L26C47

**Comment by harshlakhara** [comment]
I tried to reproduce the above issue and was able to reproduce it as well. As you can see in the screenshot below even the angular language service shows the warning for the project name. I think giving the dot in the name of the appllication should not be alllowed or it can be converted to "-" while creating the application.
![image](https://github.com/angular/angular-cli/assets/68987112/5d1911e6-2e09-40df-9879-3f756da0abce)
![image](https://github.com/angular/angular-cli/assets/68987112/8f56e335-5f16-4b2a-99b1-10ff3a889d1d)

**Comment by harshlakhara** [comment]
I have raised two PRs to fix the above issue #25704. Maintainer can either merge one of them to fix the issue. 
PR1: #26908 (This PR is to not accept the dot(.) in app name while doing "ng new")
PR2: #26909 (This PR is to validate the projectName with dot(.) in it)
